### PR TITLE
Qt6 migrations that are compatible with Qt5

### PR DESCRIPTION
**Pull request by mjcarroll**
Set of qt5+qt6 compatible changes to CMake and remove logic deprecated in qt < 5.15

### CODE REVIEW

File: src/plugins/minimal_scene/MinimalScene.hh [hunk 1]
```diff
@@ -380,101 +380,101 @@ namespace gz::gui::plugins
     /// \param[in] _view_controller The camera view controller type to set
     public: void SetCameraViewController(const std::string &_view_controller);
 
-    /// \brief Slot called when thread is ready to be started
-    public Q_SLOTS: void Ready();
+  /// \brief Slot called when thread is ready to be started
+  public Q_SLOTS: void Ready();
```
Comment (iche033) [inline src/plugins/minimal_scene/MinimalScene.hh:384]: undo indentation? looks like the original indentation is correct?

File: src/plugins/minimal_scene/MinimalScene.hh [hunk 2]
```diff
@@ -380,101 +380,101 @@ namespace gz::gui::plugins
...
+  signals: void PendingNewTexture();
+
+  /// \brief Texture size
+  public: QSize size = QSize(0, 0);
+
+  /// \brief Mutex to protect the texture variables
+  public: QMutex mutex;
+
+  /// \brief See RenderSync
+  public: RenderSync &renderSync;
```
Comment (iche033) [inline src/plugins/minimal_scene/MinimalScene.hh:477]: update indentation?

File: src/plugins/minimal_scene/MinimalScene.hh (hunk 1 above)
Comment (mjcarroll) [reply src/plugins/minimal_scene/MinimalScene.hh]: I think that the indentation may be incorrect on `main`, so I have fixed it here.

Comment (mjcarroll) [reply src/plugins/minimal_scene/MinimalScene.hh]: (Just now, so take another look)

File: src/plugins/minimal_scene/MinimalScene.hh [hunk 3]
```diff
@@ -380,108 +380,108 @@ namespace gz::gui::plugins
...
+    private: QSGNode *updatePaintNode(QSGNode *_oldNode,
+        QQuickItem::UpdatePaintNodeData *_data) override;
 
-  /// \internal
-  /// \brief Pointer to private data.
-  GZ_UTILS_UNIQUE_IMPL_PTR(dataPtr)
+    /// \internal
+    /// \brief Pointer to private data.
+    GZ_UTILS_UNIQUE_IMPL_PTR(dataPtr)
 };
```
Comment (iche033) [inline src/plugins/minimal_scene/MinimalScene.hh:431]: hmm I think there's still some minor indentation issue. How does this look:


```diff
diff --git a/src/plugins/minimal_scene/MinimalScene.hh b/src/plugins/minimal_scene/MinimalScene.hh
index 5666703e..e7edf343 100644
--- a/src/plugins/minimal_scene/MinimalScene.hh
+++ b/src/plugins/minimal_scene/MinimalScene.hh
@@ -428,11 +428,11 @@ namespace gz::gui::plugins
     /// \internal
     /// \brief Pointer to private data.
     GZ_UTILS_UNIQUE_IMPL_PTR(dataPtr)
-};
+  };
 
-/// \brief Texture node for displaying the render texture from gz-renderer
-class TextureNode : public QObject, public QSGSimpleTextureNode
-{
+  /// \brief Texture node for displaying the render texture from gz-renderer
+  class TextureNode : public QObject, public QSGSimpleTextureNode
+  {
     Q_OBJECT
 
     /// \brief Constructor
@@ -481,7 +481,7 @@ class TextureNode : public QObject, public QSGSimpleTextureNode
 
     /// \brief Pointer to render interface to handle OpenGL/Metal compatibility
     private: std::unique_ptr<TextureNodeRhi> rhi;
-};
+  };
 }  // namespace gz::gui::plugins
```

**Review by iche033 — APPROVED** [review]

**Pull request merged**